src/react-components/src/components/Button/Button.tsx

```diff
@@ -17,6 +17,7 @@ export interface ButtonOptions {
     mouseDown?: boolean;
     iconPosition?: ButtonIconPositions;
     iconName?: string;
+    iconRotation?: "rotate-180" | "rotate-270" | "rotate-0" | "rotate-90";
```

Comment:
This might be a good time to extract all the iconProps into an `iconOptions` object.  (See `Input.tsx` and how it handles `HelperErrorText`).  I say this because `iconPosition` and `iconRotation` seem like similar props, except that `IconRotation` is a prop of `Icon` and `IconPosition` is a prop handled in `Button`

Also, passing in so many props like this starts to get really frustrating, and the `-Options` objects allow for that object to be built and added to as it is passed from component to component.  

so... uh two questions.  
1) What do you think about this approach?  It's already in `Input` (and `DateRangeForm` which uses `Input`).  I'm up for any other ideas on how to wrangel
2) Is this a good place to make that switch over?

I'm happy to chat offline about this on Tuesday also!

Reply:
I updated to move the potential rotation  options into the `iconRotationOptions` enum